components/scream/src/diagnostics/tests/potential_temperature_test.cpp

```diff
@@ -0,0 +1,277 @@
+          pressure("pressure",num_mid_packs);
+  // Output views
+  view_1d theta("theta",num_mid_packs);
+
+  auto dview_as_real = [&] (const view_1d& v) -> rview_1d {
+    return rview_1d(reinterpret_cast<RealType*>(v.data()),v.size()*pack_size);
+  };
+  auto hview_as_real = [&] (const typename view_1d::HostMirror& v) -> typename rview_1d::HostMirror {
+    return typename rview_1d::HostMirror(reinterpret_cast<RealType*>(v.data()),v.size()*pack_size);
+  };
+
+  // Construct random input data
+  using RPDF = std::uniform_real_distribution<RealType>;
+  RPDF pdf_pres(0.0,PC::P0),
+       pdf_temp(200.0,400.0);
+
+  //contruct random integers
+  using IPDF = std::uniform_int_distribution<int>;
+  IPDF pdf_rand_int(1,100);
+
+  ekat::genRandArray(dview_as_real(temperature),     engine,pdf_temp);
+  ekat::genRandArray(dview_as_real(pressure),        engine,pdf_pres);
+
+  // A time stamp
+  util::TimeStamp t0 ({2022,1,1},{0,0,0});
+
+  // Construct the Diagnostic
+  ekat::ParameterList params;
+  params.set<std::string>("Diagnostic Name", "Potential Temperature");
+  params.set<std::string>("Grid", "Point Grid");
+  auto diag = std::make_shared<PotentialTemperatureDiagnostic>(comm,params);
+  diag->set_grids(gm);
+
+  // Set the required fields for the diagnostic.
+  std::map<std::string,Field> input_fields;
+  for (const auto& req : diag->get_required_field_requests()) {
+    Field f(req.fid);
+    f.allocate_view();
+    const auto name = f.name();
+    f.get_header().get_tracking().update_time_stamp(t0);
+    diag->set_required_field(f.get_const());
+    REQUIRE_THROWS(diag->set_computed_field(f));
+    input_fields.emplace(name,f);
+  }
+
+  // Initialize the diagnostic
+  diag->initialize(t0,RunType::Initial);
+
+  // Run tests
+  const ScalarT p0   = PC::P0;
+  const ScalarT zero = 0.0;
+  const ScalarT one  = 1.0;
+
+  // Get views of input data and set to random values
+  const auto& T_mid_f = input_fields["T_mid"];
+  const auto& T_mid_v = T_mid_f.get_view<ScalarT**>();
+  const auto& p_mid_f = input_fields["p_mid"];
+  const auto& p_mid_v = p_mid_f.get_view<ScalarT**>();
+
+  // Test 1 - property tests 
+  //  - theta(T=0) = 0
+  {
+  Field zero_f = T_mid_f;  // Field with only zeros
+  zero_f.deep_copy(0.0);
+  Kokkos::parallel_for("", policy, KOKKOS_LAMBDA(const MemberType& team) {
+    const int i = team.league_rank();
+    const auto& T_sub = ekat::subview(T_mid_v,i);
+    const auto& p_sub = ekat::subview(p_mid_v,i);
+    Kokkos::deep_copy(T_sub,zero);
+    Kokkos::deep_copy(p_sub,p0);
+  });
+  Kokkos::fence();
+  const auto& diag_out = diag->get_diagnostic(100.0);
+  REQUIRE(views_are_equal(diag_out,zero_f));
+  }
+  //  - theta=T when p=p0
+  {
+  Kokkos::parallel_for("", policy, KOKKOS_LAMBDA(const MemberType& team) {
+    const int i = team.league_rank();
+    const auto& T_sub = ekat::subview(T_mid_v,i);
+    const auto& p_sub = ekat::subview(p_mid_v,i);
+    Kokkos::deep_copy(T_sub,temperature);
```

Comment:
You cannot call deep_copy from inside a pfor. This should make compiler crap out (at least on GPU).

Reply:
I think you're better off doing this for loop on the host. Something like
```
const auto& T_mid_f = input_fields["T_mid"];
auto T_mid_v = T_mid_f.get_view<Real**>();
for (int icol=0; icol<ncols;++icol) {
  auto T_col = ekat::subview(T_mid_v,icol);
  Kokkos::deep_copy(T_col,temperature);
}
```